src/content/protocols/euler/euler.md

```diff
@@ -0,0 +1,637 @@
+
+### Fee Configuration
+
+The [ProtocolConfig](https://etherscan.io/address/0x4cD6BF1D183264c02Be7748Cb5cd3A47d013351b) contract controls global fee settings through `setProtocolFeeShare`, `setInterestFeeRange`, and vault-specific overrides, which affect all existing user positions immediately. Per-vault fee configuration is available through `setFeeReceiver` and `setInterestFee` on individual EVaults.
+
+### EulerEarn Vault System
+
+EulerEarn Vaults receive user funds and allocate the funds to different EVaults. The EulerEarn Vaults are governed and managed by the respective owner. The owner can assign the role-based permissions to curators and allocators. The curator manages risk of the Earn Vault strategy adding EVaults and supply caps. Allocators control and execute fund distribution to the EVaults that the curator has enabled. The [EulerEarnFactory2](https://etherscan.io/address/0x59709B029B140C853FE28d277f83C3a65e308aF4) exposes `setPerspective` to update strategy validation rules for all EulerEarn vaults.
+
+### EulerSwap
+
+The [EulerSwapFactory](https://etherscan.io/address/0xb013be1D0D380C13B58e889f412895970A2Cf228) controls swap fee configuration through `setProtocolFee` and `setProtocolFeeRecipient` for all EulerSwap pools.
+
+With the current setup, the timelock protections (48-hour and 4-day delays) fall below the 7-day minimum threshold required for medium risk classification, and numerous critical functions affecting user funds can be executed by multisigs or role holders.
+
+> Upgradeability score: High
+
+## Autonomy
+
+Euler V2 relies on external oracle providers selected by vault creators and risk curators for all price data required for collateral valuations and liquidation decisions. Vault creators are free to choose appropriate price oracle implementations from various providers including Chainlink, Pyth, RedStone, Chronicle, Lido, Pendle, Uniswap V3, Midas, MEV Capital, Idle, and Resolv, thus delegating responsibility to individual actors.
+
+The protocol employs CrossAdapters that derive prices through sequential multiplication of two external oracles sharing a base or quote asset. These create dependency chains where the failure of either component oracle causes complete price feed failure. For ERC-4626 vault collaterals, the system depends on external vaults' `convertToAssets()` functions, which may be vulnerable to manipulation if those vaults lack adequate protections.
+
+Pull-based oracles like Pyth introduce operational dependencies where transactions can fail if price updates are not fetched within the 2-3 minute validity window. Observable errors include intermittent failures when querying vault data and operations failing unpredictably based on timing, highlighting the practical risks of external oracle dependencies.
+
+> Autonomy score: High
+
+## Exit Window
```

Comment:
go through the upgradeability vectors 1-5 (see excalidraw) and write that the exit window is below 7 days, but currently 4. Additionally for vector 2,3 and 5 write that 3rd party management can be even without exit window and it depends on the private entity controlling this vector